src/satosa/frontends/saml2.py

```diff
@@ -265,93 +276,86 @@ def _handle_authn_response(self, context, internal_response, idp):
         request_state = self.load_state(context.state)
 
         resp_args = request_state["resp_args"]
-        internal_response.attributes = self._filter_attributes(idp, internal_response, context)
-        ava = self.converter.from_internal(self.attribute_profile, internal_response.attributes)
+        sp_entity_id = resp_args["sp_entity_id"]
+        internal_response.attributes = self._filter_attributes(
+            idp, internal_response, context)
+        ava = self.converter.from_internal(
+            self.attribute_profile, internal_response.attributes)
 
         auth_info = {}
         if self.acr_mapping:
-            auth_info["class_ref"] = self.acr_mapping.get(internal_response.auth_info.issuer, self.acr_mapping[""])
+            auth_info["class_ref"] = self.acr_mapping.get(
+                internal_response.auth_info.issuer, self.acr_mapping[""])
```

Comment:
I think we could switch to `get_dict_defaults()` here, instead of manually trying the `""` key. 
I think this happens elsewhere too, and we can agree that the `get_dict_defaults` should dictate on how fallback keys should be checked (as a reminder there are two special/fallback keys: `""` and `"default"`).